website/content/docs/upgrading/identity-deduplication.mdx

```diff
@@ -0,0 +1,406 @@
+  name              = "BOB"
+  policies          = ["TEST"]
+}
+
+
+resource "vault_identity_entity" "bob" {
+    name              = "bob"
+    policies          = ["test"]
+}
+```
+
+By default this Terraform config would not work correctly since Vault would
+match the names in a case-insensitive way and reject creating the second
+resource. But if you have had duplicates reported in your Vault cluster, that
+means that due to some historical issue Vault is running in a mode that would
+allow both of these case variations to exist a separate entities.
+
+After activating the `force-identity-deduplication` flag, one of these entities
+will no longer have the specified name and so Terraform will attempt to rename
+it and be unable to because that violates the case-insensitive name constraint
+that has be re-introduced. An apply would result in an error such as:
+
+```
+➜  tf_dupe_testing terraform apply
+vault_identity_entity.bob: Refreshing state... [id=e8c5e633-fe37-5a49-4a29-32e2643d03bd]
+vault_identity_entity.BOB: Refreshing state... [id=2577bc3f-67ab-dab7-93dc-e86f78194ff0]
+
+Terraform used the selected providers to generate the following execution plan. Resource
+actions are indicated with the following symbols:
+  ~ update in-place
+
+Terraform will perform the following actions:
+
+  # vault_identity_entity.bob will be updated in-place
+  ~ resource "vault_identity_entity" "bob" {
+      + external_policies = false
+        id                = "e8c5e633-fe37-5a49-4a29-32e2643d03bd"
+      ~ name              = "bob-e8c5e633-fe37-5a49-4a29-32e2643d03bd" -> "bob"
+        # (3 unchanged attributes hidden)
+    }
+
+Plan: 0 to add, 1 to change, 0 to destroy.
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+
+vault_identity_entity.bob: Modifying... [id=e8c5e633-fe37-5a49-4a29-32e2643d03bd]
+╷
+│ Error: error updating IdentityEntity "e8c5e633-fe37-5a49-4a29-32e2643d03bd": Error making API request.
+│
+│ URL: PUT https://127.0.0.1:8200/v1/identity/entity/id/e8c5e633-fe37-5a49-4a29-32e2643d03bd
+│ Code: 400. Errors:
+│
+│ * entity name is already in use
+│
+│   with vault_identity_entity.bob,
+│   on main.tf line 17, in resource "vault_identity_entity" "bob":
+│   17: resource "vault_identity_entity" "bob" {
+│
+╵
+```
+
+To resolve this, you will need to update any Terraform configuration with the
+new name which you can get from the previous duplicate report:
+
+```
+WARN: [...] force_deduplication="would rename to bob-5789a392-e4fa-4df3-b2a9-ec11839f7ed1"
+```
+
+Or from future unseal logs:
+
+```
+WARN: duplicate entity `bob` was renamed to `bob-5789a392-e4fa-4df3-b2a9-ec11839f7ed1`
+```
+
+The same idea would apply to any other external systems that reference an entity
+or group by name. Simply changing the reference to use the new name is the
+simplest resolution.
+
+### Activating the `force-identity-deduplication` flag
```

Comment:
I know it was said somewhere way up above, but once you write this part I think it might be good to make sure there's another warning here that this is a one-time thing, so they should review the resolution information for their relevant duplicate types above before activating the flag.